lib/compress.js

```diff
@@ -2989,33 +2989,30 @@ merge(Compressor.prototype, {
         return self;
     });
 
-    function has_side_effects_or_prop_access(node, compressor) {
-        var save_pure_getters = compressor.option("pure_getters");
-        compressor.options.pure_getters = false;
-        var ret = node.has_side_effects(compressor);
-        compressor.options.pure_getters = save_pure_getters;
-        return ret;
-    }
```

Comment:
This complication is unnecessary because (in ES5) LHS of an assignment [can only be `AST_Symbol` or `AST_PropAccess`](https://github.com/mishoo/UglifyJS2/blob/cf45e2f79b543ebae60c5de54166b20da4522c25/lib/parse.js#L1503-L1507)

And by `AST_Symbol` the only case applicable here is `AST_SymbolRef`.